lib/service/kinveyCompletionHandler.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+function createCompletionHandler(task, callback) {
+  function completionHandler(entity) {
+    if (!entity) entity = {};
+
+    let api = {};
```

Comment:
I'm wondering if it would be better to define these things inline.  For example here:  

```

task.response.body = entity;

const api = {

  responseCallback: callback,
  result: task.response,
  
  created() { 
    result.statusCode = 201;
    return this;
  }, 
  accepted() {
    result.statusCode = 202;
    return this;
  },
  // rest of methods go here
};
```

WDYT?  @vishwakrishnamurthy? @sjl2024?